app/controllers/additionalInformation/index/RemoveAdditionalInformationYesNoController.scala

```diff
@@ -51,6 +52,15 @@ class RemoveAdditionalInformationYesNoController @Inject() (
     controllers.additionalInformation.routes.AddAnotherAdditionalInformationController
       .onPageLoad(lrn, mode)
 
+  private def addRoute(itemsCount: Int, lrn: LocalReferenceNumber, mode: Mode): Call =
+    if (itemsCount > 0) {
+      controllers.additionalInformation.routes.AddAnotherAdditionalInformationController
+        .onPageLoad(lrn, mode)
+    } else {
+      controllers.additionalInformation.routes.AddAdditionalInformationYesNoController
+        .onPageLoad(lrn, mode)
+    }
```

Comment:
Ordinarily we would do this check in the onPageLoad of `AddAnotherAdditionalInformationController`. If there aren't any additional informations we redirect them to `AddAdditionalInformationYesNoController`